### ISSUE TITLE
Code snippets use wrong palette in explicit light mode with dark system preference

### ISSUE THREAD

**Issue by laymonage** [django/djangoproject.com#1966]
## Bug

If you:
- set the theme to light explicitly, i.e. the Sun icon is displayed and `data-theme="light"` is set on the `<html>`
- have a "dark" color scheme preference set on your system

The code snippets use the wrong color palette:

<img width="886" alt="Image" src="https://github.com/user-attachments/assets/ecbb1231-a12e-419d-83dd-9cad3ac42e6d" />

Example page: https://docs.djangoproject.com/en/5.1/topics/db/models/#inheritance-and-reverse-relations

## Expected

The light color palette should be used:

<img width="886" alt="Image" src="https://github.com/user-attachments/assets/c06ab280-8e88-403a-aa0d-b2bdb975e0ad" />

## Cause

The `_dark-pygments.scss` styles set the dark color palette using a media query regardless of what the `data-theme` is set to:

https://github.com/django/djangoproject.com/blob/f6a6d91e4704b7d47ce3f74513268a6ce972bb28/djangoproject/scss/_dark-pygments.scss#L2-L3

The media query takes higher precedence than the default light theme. The selector should specifically check for `:root[data-theme="auto"]`, so that the styles don't get applied when the media query matches but the theme isn't `auto`.

**Comment by bmispelon** [comment]
Nice catch, I can confirm that I see the behavior in my browser too (Firefox 😞 )

**Comment by pauloxnet** [comment]
I confirm the bug too. Nice catch.
